x/mint/keeper/keeper_test.go

```diff
@@ -719,3 +719,348 @@ func (suite *KeeperTestSuite) TestDistributeDeveloperRewards() {
 		})
 	}
 }
+
+// TestCalculateTruncationsDelta tests that the per-epoch truncation delta is calculated
+// correctly. That includes checking that the pre-existing delta is
+// extracted from the store at the desired index and summed with the new delta
+// from the input values.
+func (suite *KeeperTestSuite) TestCalculateTruncationDelta() {
+	const mintDenom = sdk.DefaultBondDenom
+	tests := map[string]struct {
+		preExistingStoreDelta sdk.Dec
+
+		moduleAccountName string
+		provisions        sdk.Dec
+		amountDistributed sdk.Int
+
+		expectedTruncationDelta sdk.Dec
+
+		expectError bool
+	}{
+		"100.6 - 100 = 0.6; no pre-existing delta; inflation delta module account": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(6, 1),
+		},
+		"100.6 - 100  + 0.3 = 0.9; pre-existing delta of 0.3; inflation delta module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(3, 1),
+
+			moduleAccountName: types.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(9, 1),
+		},
+		"100.6 - 100  + 0.5 = 1.1; pre-existing delta of 1.1; inflation delta module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(5, 1),
+
+			moduleAccountName: types.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(11, 1),
+		},
+		"82191.178 - 17808  + 33.33 = 64416508; pre-existing delta of 33.33; developer vesting module account": {
+			preExistingStoreDelta: sdk.NewDecWithPrec(3333, 2),
+
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			provisions:        sdk.NewDecWithPrec(82191178, 3),
+			amountDistributed: sdk.NewInt(17808),
+
+			expectedTruncationDelta: sdk.NewDecWithPrec(64416508, 3),
+		},
+		"attempt to use invalid module account name - panic": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectError: true,
+		},
+		"invalid module account - error": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			provisions:        sdk.NewDecWithPrec(1006, 1),
+			amountDistributed: sdk.NewInt(100),
+
+			expectError: true,
+		},
+	}
+	for name, tc := range tests {
```

Comment:
I think we absolutely need to test a case where the overflow of the accumulator happens over greater than one epoch. In other words: pre-existing delta of 0.1, and the next two epochs have a delta of 0.4

Reply:
I'm not sure why we need that? What we are testing here is that the math is correct based on the input parameters and the value of the store index. I think the existing tests cover this pretty well.

Testing this over several epoch in such as way is the same as judging some of the existing test cases separately

Reply:
I guess my concern is the store getting prematurely overridden before reaching > 1 if the store takes multiple epochs to reach this value. I felt like having the check there would be nice to have, especially if any of this logic gets touched in the future. On my second pass I will check how necessary I feel this should actually be

Reply:
Sorry, I'm not sure I follow how this concern specifically relates to a unit test for the `calculateTotalTruncationDelta` function.

This is something that is covered by a functional test that runs the `AfterEpochEnd` hook several times and ensures that the amounts are valid:
https://github.com/osmosis-labs/osmosis/blob/d7caed8acfe3b756134c42c839bfca1d6225cb1b/x/mint/keeper/hooks_test.go#L444

Reply:
Let's say there is a bug in `calculateTotalTruncationDelta` which causes the retained value to only be retained for one epoch, then, if the epoch after that still doesn't push it over the "1" delta to be distributed, it gets reset to zero. This case would not be caught by higher abstraction levels right?

This isn't a huge deal at all, I just thought it was something that should be added to these test cases but it isn't a blocker

Reply:
It should be caught in the linked test, and I don't see how this could be tested here since `calculateTotalTruncationDelta` doesn't deal with any epoch logic.

Reply:
That's fair! Just wanted to make sure we had test coverage somewhere for this scenario!